app/views/layouts/application.html.erb

```diff
@@ -6,7 +6,7 @@
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
-<body>
+<body class="<%= controller.controller_name %> <%= controller.action_name %>">
```

Comment:
The above suggestion would make this unnecessary. Also, I think in general it's probably best to have classes that represent types of things that cut across pages, or describe their purpose rather than origin, removing the need for selecting on the controller/action. This would make them more portable, and allow the same javascript to manipulate any page that thing can come on.

That said, I think this is a really cool idea for the rare case that a hook might really be the best option, or even to help with styling a page very slightly differently than other pages. It might also be a nice way to scope actions of buttons that are hooked in at the document level... thus something like `$(document).on "click", ".save-item", -> ...` would become `$(document).on "click", "body.items .save-item", -> ...`, which would reduce possible accidental bugs due to classes that end up the same unintentionally.